src/build_cfg.c

```diff
@@ -415,25 +415,27 @@ static const LocalVariable *build_address_of_expression(struct State *st, const
     assert(0);
 }
 
-static const LocalVariable *build_function_call(struct State *st, const AstExpression *expr)
+static const LocalVariable *build_function_call(struct State *st, const Location location, const AstCall *call, const LocalVariable *self, const Type *returntype)
 {
-    assert(expr->kind == AST_EXPR_FUNCTION_CALL);
+    const LocalVariable **args = calloc(call->nargs + 2, sizeof(args[0]));  // NOLINT
+    int k = 0;
+    if (self)
+        args[k++] = self;
```

Comment:
Ideally the self argument could go anywhere, e.g. `fread()` takes it as the last argument.